aten/src/TH/THTensor.hpp

```diff
@@ -33,11 +33,34 @@ inline THStorage* THTensor_getStoragePtr(const THTensor* tensor) {
   return tensor->storage_.unsafeGetStorageImpl();
 }
 
+inline void THTensor_refreshContiguous(THTensor *self)
+{
+  self->is_contiguous_ = 1;
+  if (self->is_empty()) return;
+  int64_t z = 1;
+  int d;
+  for(d = self->dim()-1; d >= 0; d--)
```

Comment:
Might as well take some time to clean up this C. You can say `for (auto d = ...`